src/service/user.service.ts

```diff
@@ -22,7 +26,12 @@ export class UserService {
         if (existingUserWithUserName) {
             throw new Error("Username already exists");
         }
-  
+
+        const emailVerificationToken = crypto.randomBytes(32).toString('hex');
+        const emailVerificationTokenExpiration = new Date(
+          Date.now() + 24 * 60 * 60 * 1000,
```

Comment:
If this token expires and they haven't verified, will they be allowed to access the app? How can they get another link when this expires?

Reply:
There were some features that they would not be able to access, but I think I it would be nice If they can not log in at all, which is what I am going to add, and also I am going to improve so that the users can be able to re-request a verification link in case the one they had is expired.